corfudb-tools/src/main/java/org/corfudb/browser/CorfuStoreBrowser.java

```diff
@@ -266,6 +274,114 @@ public int dropTable(String namespace, String tablename) {
         return tableSize;
     }
 
+    public int editRecord(String namespace, String tableName,
+        String keyToEdit, String newRecord) {
+        verifyNamespaceAndTablename(namespace, tableName);
+
+        if (keyToEdit == null) {
+            throw new IllegalArgumentException("Please specify the key to " +
+                "edit --keyToEdit");
+        }
+
+        if (newRecord == null) {
+            throw new IllegalArgumentException("Please specify the newRecord " +
+                "--newRecord");
+        }
+
+        log.info("\n======================\n");
+        String fullName = TableRegistry.getFullyQualifiedTableName(namespace,
+            tableName);
+        UUID streamUUID = CorfuRuntime.getStreamID(fullName);
+        CorfuTable<CorfuDynamicKey, CorfuDynamicRecord> table = getTable(
+            namespace, tableName);
+
+        TableName tableNameProto = TableName.newBuilder().setTableName(tableName)
+            .setNamespace(namespace).build();
+
+        CorfuStoreMetadata.KeyValueDefaultRecord keyValueDefaultRecord =
+            runtime.getTableRegistry().getTableNameToKeyValueTypeTable().get(tableNameProto);
+        Any defaultKeyAny = keyValueDefaultRecord.getKey();
+        Any defaultValueAny = keyValueDefaultRecord.getValue();
+
+        DynamicMessage keyMsg = createMsgFromJson(defaultKeyAny, keyToEdit);
+        DynamicMessage newValueMsg = createMsgFromJson(defaultValueAny,
+            newRecord);
+
+        if (keyMsg == null || newValueMsg == null) {
+            return 0;
+        }
+
+        CorfuDynamicKey dynamicKey =
+            new CorfuDynamicKey(defaultKeyAny.getTypeUrl(), keyMsg);
+
+        if (table.containsKey(dynamicKey)) {
+            log.info("Editing record with Key {} in table {} and " +
+                    "namespace {} with new record {}.  Stream Id {}.",
+                keyToEdit, tableName, namespace, newRecord, streamUUID);
+            String payloadTypeUrl = table.get(dynamicKey).getPayloadTypeUrl();
+            String metadataTypeUrl = table.get(dynamicKey).getMetadataTypeUrl();
+            DynamicMessage metadata = table.get(dynamicKey).getMetadata();
```

Comment:
Does table allow null as a key's value? If so, we might hit NPE

Reply:
we construct the key explicitly so it will not be null.

Reply:
Sorry for the confusion, I mean "table.get(dynamicKey)" might be null?